lib/fastlane/plugin/wpmreleasetoolkit/helper/app_size_metrics_helper.rb

```diff
@@ -0,0 +1,95 @@
+require 'json'
+require 'net/http'
+require 'zlib'
+
+module Fastlane
+  module WPMRT
+    # A helper class to build an App Size Metrics payload and send it to a server (or write it to disk)
+    #
+    # The payload generated (and sent) by this helper conforms to the API for grouped metrics described in
+    # https://github.com/Automattic/apps-metrics
+    #
+    class AppSizeMetricsHelper
+      # @param [Hash] group_meta Metadata common to all the metrics. Can be any arbitrary set of key/value pairs.
+      #
+      def initialize(group_meta = {})
+        self.meta = group_meta
+        @metrics = []
+      end
+
+      # Sets the metadata common to the whole group of metrics in the payload being built by this helper instance
+      #
+      # @param [Hash] hash The metadata common to all the metrics of the payload built by that helper instance. Can be any arbitrary set of key/value pairs
+      #
+      def meta=(hash)
+        @meta = (hash.compact || {}).map { |key, value| { name: key.to_s, value: value } }
+      end
+
+      # Adds a single metric to the group of metrics
+      #
+      # @param [String] name The metric name
+      # @param [Integer] value The metric value
+      # @param [Hash] meta The arbitrary dictionary of metadata to associate to that metric entry
+      #
+      def add_metric(name:, value:, meta: nil)
+        metric = { name: name, value: value }
+        meta = (meta || {}).compact # Remove nil values if any
```

Comment:
Thinking out loud: How would this code feel if there was a `remove_nil_values_if_any(meta)` call instead?

```ruby
def remove_nil_values_if_any(hash)
  (hash || {}).compact
end
```

The rationale for this suggestion comes from one of my favorite bits from ["Clean Code" by Uncle Bob](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882):

<img width="500" alt="Screen Shot 2019-10-31 at 9 23 28 am" src="https://user-images.githubusercontent.com/1218433/67903657-4a477380-fbc0-11e9-8f6a-e3c3f210f92a.png">

Reply:
👍